src/wasteApp_server/src/main/java/com/fantastiSquad/wasteApp/controllers/CityController.java

```diff
@@ -0,0 +1,97 @@
+package com.fantastiSquad.wasteApp.controllers;
+
+import com.fantastiSquad.wasteApp.models.entities.City;
+
+import com.fantastiSquad.wasteApp.models.services.CityService;
+import java.util.List;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@CrossOrigin("*")
+@RestController
+@RequestMapping("/city")
+public class CityController {
+
+  @Autowired
+  CityService cityService;
+
+  @GetMapping(value = "")
+  public List<City> getAllCities() {
+    return cityService.getAllCities();
+  }
+
+  @GetMapping(value = "/{id}")
+  public ResponseEntity<City> getCityById(@PathVariable("id") Long id) {
+    Optional<City> cityFromDb = cityService.getCityById(id);
+    if (cityFromDb.isPresent()) {
+      return new ResponseEntity<City>(cityFromDb.get(), HttpStatus.OK);
+    } else {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+          "La ville avec l'id " + id + " n'a pas été trouvée!");
+    }
+  }
+
+  @GetMapping(value="/searchCity/{name}")
+  public ResponseEntity<List<City>> getCityByName(@PathVariable("name") String name){
+      String nameToModify = name;
+      List<City> cityFromDb = cityService.searchCityByName(nameToModify.toLowerCase()).get();
+
+      if(!cityFromDb.isEmpty()){
+        return new ResponseEntity<List<City>>(cityFromDb, HttpStatus.OK);
+      } else {
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Aucune ville trouvée avec ce nom");
+      }
+  }
+  
+  @PostMapping(value="/create")
```

Comment:
/create not needed since url type is Get.

Reply:
done